src/internal/operations/backup.go

```diff
@@ -307,10 +319,25 @@ func useIncrementalBackup(sel selectors.Selector, opts control.Options) bool {
 // Producer funcs
 // ---------------------------------------------------------------------------
 
+type BackupProducer interface {
```

Comment:
Although I see what you're trying to do with Kopia being a RestoreConsumer...

Reply:
> Kopia being a RestoreConsumer

Close.
* Backup: GC produces (BackupProducer), Kopia consumes (BackupConsumer)
* Restore: Kopia produces (RestoreProducer), GC consumes (RestoreConsumer)

> probably live in connector

Hah, in another PR, ash moved ownership into kopia.  Now you two have to fight for it.

Imo, operations is the owner (though the interfaces should move into a different package).  GC and Kopia should be compliant with operations interfaces, which will make them easy to swap out for mocks or alternative production/consumption systems.

Reply:
I agree with moving them to a different package. Whether that is a subpackage of operation or of connector/kopia I'm not real picky. What I would like to see is stuff like `var _ BackupProducer = &GraphConnector` so that we actually know we implement the interface we think we do

The above check matters slightly less because we're already using the implementations, but is a good way to catch missed changes for stuff that may not be used yet (i.e. feature buildout or something)

Reply:
Done.  All interfaces have been moved to `operations/inject`.